### ISSUE TITLE
Zip dependency is funky

### ISSUE THREAD

**Issue by Toma400** [dariolah/odsreader#1]
So, basically there are two things related to each other partially, but not exactly.

1. Nimble file doesn't state `zip` as requirement, meaning Nimble installation won't download it automatically - meaning unless you notice that, you will get errors of unknown import
2. But then, for some reason `zip` dependency also produces the error:
```
CC: D:/Ministerstwo Tec niczne/Nim/nim-2.C:\Users\Ryzen 5\.nimble\pkgs2\zip-0.3.1-63d994f02253805ec10bfb383c456809f2303345\zip\private\libzip_all.c:34:10:0. /lib/std/oserrors.nim
fatal error: zlib.h: No such file or directory
   34 | #include <zlib.h>
      |          ^~~~~~~~
compilation termiCnC: D:/Minisaterstwto eTechniczne/Nim/nim-2.0.0/lib/std/private/oscommon.nimd
.
CC: D:/Ministerstwo Techniczne/Nim/nim-2.0.0/lib/std/private/osfiles.nim
CC: D:/Ministerstwo Techniczne/Nim/nim-2.0.0/lib/std/private/win_setenv.nim
CC: D:/Ministerstwo Techniczne/Nim/nim-2.0.0/lib/std/cmdline.nim
Error: execution of an external compiler program 'gcc.exe -c  -w -fmax-errors=3 -mno-ms-bitfields -DWIN32_LEAN_AND_MEAN   -I"D:\Ministerstwo Techniczne\Nim\nim-2.0.0\lib" -I"F:\[archiver]\[on pulpit]\komórka\PDFy\Arkusze kalkulacyjne" -o "C:\Users\Ryzen 5\nimcache\test_d\@mC@c@sUsers@sRyzen 5@s.nimble@spkgs2@szip-0.3.1-63d994f02253805ec10bfb383c456809f2303345@szip@sprivate@slibzip_all.c.o" "C:\Users\Ryzen 5\.nimble\pkgs2\zip-0.3.1-63d994f02253805ec10bfb383c456809f2303345\zip\private\libzip_all.c"' failed with exit code: 1
```
From what I gathered on Nim Discord server, it is missing file from development build? Not sure how I can even install it, but I think by default I shouldn't expect to try failproof library, so I think getting this to your attention would be good - just so you can maybe adjust the code, so either it doesn't involve development files (or whatever error above says) or that it does download it alongside `zip` library that should be included in Nimble anyway

Thank you for making this library! ^^

**Comment by dariolah** [comment]
Library is not tested on Windows platform.

Looks like that on Windows `zip/zipfiles` doesn't use pre-compiled library, but compiles `zip/private/libzip_all.c`. This C code depends on `zlib` library. You have to have `zlib` development files (.h, .a, .so) as well somewhere in gcc include path.

Maybe ask on Nim forum how to add ZLIB to Nim toolchain on Windows (MinGW). Copy files inside `lib/`, `include/` dirs?

http://www.zlib.net/

**Comment by Toma400** [comment]
Yeah, I've seen this issue is prevailing on the repository for years now and noone really bothered making PR to it. Funnily enough I noticed the code for your lib can be very easily repurposed for `zippy` - I just edited it slightly and it seems to work? Another fun thing is that zippy is pure Nim library, so it cuts the C dependency entirely.
If you don't mind, I will do the fork and then contribute the `zip > zippy` migration as PR - you are free to reject it, but if you don't, I will be very happy, haha ^^
